torch/csrc/cuda/memory_snapshot.cpp

```diff
@@ -395,6 +402,7 @@ std::string _memory_snapshot_pickled() {
   result.insert("segments", segments);
   result.insert("device_traces", traces);
   result.insert("allocator_settings", allocator_settings);
+  result.insert("user_defined", user_defined);
```

Comment:
nit: maybe `external_annotations`, otherwise I ask "used-defined what?"

Reply:
That makes sense! Changed